src/applications/personalization/dashboard/tests/e2e/confirm-email-address.cypress.spec.js

```diff
@@ -0,0 +1,165 @@
+    cy.intercept('/v0/profile/full_name', fullName).as('nameB');
+    cy.intercept('GET', '/v0/feature_toggles*', {
+      data: {
+        features: [
+          {
+            name: 'mhvEmailConfirmation',
+            value: true,
+          },
+        ],
+      },
+    });
+    mockLocalStorage();
+
+    cy.login(mockUser);
+    cy.visit('my-va/');
+  });
+
+  it('should show the error alert when confirm fails and handle a successful retry attempt', () => {
+    let hasFailed = false;
+    cy.intercept('PUT', '/v0/profile/email_addresses', req => {
+      if (!hasFailed) {
+        hasFailed = true;
+        req.reply(buildUpdateEmailResponse(false));
+      } else {
+        req.reply(buildUpdateEmailResponse());
+      }
+    }).as('updateEmail');
+
+    clickConfirmEmail();
+    cy.wait('@updateEmail');
+
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
+      'not.exist',
+    );
+    cy.findByTestId('mhv-alert--confirm-error').should('be.visible');
+    cy.findByTestId('mhv-alert--confirm-error').should('be.focused');
+
+    cy.getCookie(MHV_EMAIL_CONFIRMATION_DISMISSED_COOKIE).should('be.null');
+
+    clickConfirmEmail();
+    cy.wait('@updateEmail');
+
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
+      'not.exist',
+    );
+    cy.get('[data-testid="mhv-alert--confirm-error"]').should('not.exist');
+    cy.findByTestId('mhv-alert--confirm-success').should('be.visible');
+    cy.findByTestId('mhv-alert--confirm-success').should('be.focused');
+    cy.getCookie(MHV_EMAIL_CONFIRMATION_DISMISSED_COOKIE).should('not.be.null');
+
+    cy.injectAxeThenAxeCheck();
+  });
+
+  it('should show the error alert when confirm fails and handle clicking the edit link', () => {
+    cy.intercept(
+      'PUT',
+      '/v0/profile/email_addresses',
+      buildUpdateEmailResponse(false),
+    ).as('updateEmail');
+
+    clickConfirmEmail();
+    cy.wait('@updateEmail');
+
+    // Verify error alert is shown and cookie isn't set.
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
+      'not.exist',
+    );
+    cy.findByTestId('mhv-alert--confirm-error').should('be.visible');
+    cy.findByTestId('mhv-alert--confirm-error').should('be.focused');
+    cy.getCookie(MHV_EMAIL_CONFIRMATION_DISMISSED_COOKIE).should('be.null');
+
+    // Click the edit link and verify redirection to contact info page.
+    clickEditEmailLink();
+    cy.url().should('include', '/profile/contact-information');
+    cy.hash().should('equal', '#contact-email-address');
+
+    cy.injectAxeThenAxeCheck();
+  });
+
+  it('should show the success alert when confirm succeeds', () => {
+    cy.intercept(
+      'PUT',
+      '/v0/profile/email_addresses',
+      buildUpdateEmailResponse(),
+    ).as('updateEmail');
+
+    clickConfirmEmail();
+    cy.wait('@updateEmail');
+
+    // Verify success alert is shown and cookie is set.
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
```

Comment:
This needs to be a find as the page may not be rendered yet even with the cy.wait above.